tidings/utils.py

```diff
@@ -58,17 +20,32 @@ def collate(*iterables, **kwargs):
     descending order rather than ascending.
 
     """
-    # TODO: Liberate into the stdlib.
     key = kwargs.pop('key', lambda a: a)
     reverse = kwargs.pop('reverse', False)
-
     min_or_max = max if reverse else min
-    peekables = [peekable(it) for it in iterables]
```

Comment:
I'll throw this in my hopper to look at when I have some more time. It might be weeks, so don't let me block if you don't have the time.